sdk/servicebus/service-bus/src/core/receiverHelper.ts

```diff
@@ -135,10 +136,52 @@ export class ReceiverHelper {
       receiver.drainCredit();
     });
 
-    return drainPromise;
+    return timeboxedPromise(drainPromise, 1000, async () => {
```

Comment:
Couple of questions, shouldn't the drainPromise itself be abortable? How was 1000 chosen, is it agreed upon in all languages? Should it be defined as a constant somewhere?

Reply:
You are right, since the promise is created here, not passed to us.  I think a `setTimeout()/clearTimeout()` pair is simpler and we don't have to deal with abort errors.

Reply:
Chris was using `200ms` based on experiments in his old PR to resolve draining when receiving messages.  I don't think other languages drains credits in these scenarios. I can use the same 200 ms.